cql3/functions/aggregate_fcts.cc

```diff
@@ -183,15 +225,62 @@ class impl_avg_function_for final : public aggregate_function::aggregate {
         ++_count;
         _sum += value_cast<Type>(data_type_for<Type>()->deserialize(*values[0]));
     }
+    virtual void set_accumultor(const opt_bytes& acc) override {
+        /*data_type tuple_type = tuple_type_impl::get_instance({data_type_for<Type>, long_type});
+        auto tuple = value_cast<tuple_type_impl::native_type>(tuple_type->deserialize(bytes_view(*acc)));
+
+        _sum = value_cast<Type>(tuple[0]);
+        _count = value_cast<int64_t>(tuple[1]);*/
+    }
+    virtual opt_bytes get_accumulator() override {
+        /*data_value tuple_val = make_tuple_value(tuple_type_impl::get_instance({data_type_for<Type>, long_type}), {data_value(_sum), data_value(_count)});
+        return tuple_val.serialize();*/
+        return std::nullopt;
+    }
+    virtual void reduce(cql_serialization_format sf, const opt_bytes& acc) override {
+        /*data_type tuple_type = tuple_type_impl::get_instance({data_type_for<Type>, long_type});
+        auto tuple = value_cast<tuple_type_impl::native_type>(tuple_type->deserialize(bytes_view(*acc)));
+
+        _sum += value_cast<Type>(tuple[0]);
+        _count += value_cast<int64_t>(tuple[1]);*/
+    }
 };
 
+
+/*template <typename Type>
+class impl_reducible_avg_function : public impl_avg_function_for<Type> {
+public:
+    virtual bytes_opt compute(cql_serialization_format sf) override {
+        data_value tuple_val = make_tuple_value(tuple_type_impl::get_instance({data_type_for<Type>, long_type}), {data_value(this->_sum), data_value(this->_count)});
+        return tuple_val.serialize();
+    }
+};*/
+
```

Comment:
What's this?

Will we have both reducible and non-reducible average functions? Why?

Reply:
There is need for aggregate query to return its partial result (without applying `compute()` on final accumulator), so the simplest way was to create reducible counterpart of `aggregate_function` which `compute()` returns the accumulator. Other way to achieve this would be to dig into `result_set_builder` and `selectors`.

Reply:
I see.

Reply:
Please add comments explaining this in the aggregate base class.

Reply:
Updated commend for `reducible_aggregate_function()` function